t/file.t

```diff
@@ -3,63 +3,88 @@
-	           "!/foo/bar",  "!\\foo\\bar",   "!foo:bar", ],
-[ "file:///foo/bar",
-	           "/foo/bar",   "\\foo\\bar",    "!foo:bar", ],
-[ "file:/foo/bar", "!/foo/bar",  "!\\foo\\bar",   "foo:bar", ],
-[ "foo/bar",       "foo/bar",    "foo\\bar",      ":foo:bar",],
-[ "file://foo3445x/bar","!//foo3445x/bar", "!\\\\foo3445x\\bar", "!foo3445x:bar"],
-[ "file://a:/",    "!//a:/",     "!A:\\",         undef],
-[ "file:///A:/",   "/A:/",       "A:\\",          undef],
-[ "file:///",      "/",          "\\",            undef],
-[ ".",             ".",          ".",             ":"],
-[ "..",            "..",         "..",            "::"],
-[ "%2E",           "!.",         "!.",           ":."],
-[ "../%2E%2E",     "!../..",     "!..\\..",      "::.."],
-);
-
-my @os = @{shift @tests};
-shift @os;  # file
-
-my $num = @tests;
-print "1..$num\n";
-
-my $testno = 1;
-
-for my $t (@tests) {
-   my @t = @$t;
-   my $file = shift @t;
-   my $err;
-
-   my $u = URI->new($file, "file");
-   my $i = 0;
-   for my $os (@os) {
-       my $f = $u->file($os);
-       my $expect = $t[$i];
-       $f = "<undef>" unless defined $f;
-       $expect = "<undef>" unless defined $expect;
-       my $loose;
-       $loose++ if $expect =~ s/^!//;
-       if ($expect ne $f) {
-           print "URI->new('$file', 'file')->file('$os') ne $expect, but $f\n";
-           $err++;
-       }
-       if (defined($t[$i]) && !$loose) {
-	   my $u2 = URI::file->new($t[$i], $os);
-           unless ($u2->as_string eq $file) {
-              print "URI::file->new('$t[$i]', '$os') ne $file, but $u2\n";
-              $err++;
-           }
-       }
-       $i++;
-   }
-   print "not " if $err;
-   print "ok $testno\n";
-   $testno++;
+
+subtest 'OS related tests (unix, win32, mac)' => sub {
+
+  my @tests =  (
+                [ "file",          "unix",       "win32",         "mac" ],
+                #----------------  ------------  ---------------  --------------
+                [ "file://localhost/foo/bar",
+                  "!/foo/bar",  "!\\foo\\bar",   "!foo:bar", ],
+                [ "file:///foo/bar",
+                  "/foo/bar",   "\\foo\\bar",    "!foo:bar", ],
+                [ "file:/foo/bar", "!/foo/bar",  "!\\foo\\bar",   "foo:bar", ],
+                [ "foo/bar",       "foo/bar",    "foo\\bar",      ":foo:bar",],
+                [ "file://foo3445x/bar","!//foo3445x/bar", "!\\\\foo3445x\\bar", "!foo3445x:bar"],
+                [ "file://a:/",    "!//a:/",     "!A:\\",         undef],
+                [ "file:///A:/",   "/A:/",       "A:\\",          undef],
+                [ "file:///",      "/",          "\\",            undef],
+                [ ".",             ".",          ".",             ":"],
+                [ "..",            "..",         "..",            "::"],
+                [ "%2E",           "!.",         "!.",           ":."],
+                [ "../%2E%2E",     "!../..",     "!..\\..",      "::.."],
+               );
+
```

Comment:
Since this file is seeing a lot of changes, could you run `perltidy` on it? That should help with making the layout readable. Also that should give us consistent indentation. I'm seeing two spaces here, but in other files we use 4 spaces. I'd like to standardize on 4 spaces.

Reply:
I ran perltidy on it. Due to the added level of `subtest`s, the table is wrapped partially.